<H1> Group 22: Project title <H1>

**Introduction**<br>
We have all seen the rise in number of forest fires in the past few years and scinetists and climate activists alike all over the world have raised their concerns regarding the same. While natural in few ecosystems, these recent forest fires root their cause to the global rise in temperature and poor land management by authorities. Due to this no matter whether a wildfire's origin is due to human intervention or natural, the drier climate makes it easy for it to spread over a region quite intensely. 
<br>
These fires are clear indicators that for change because not only do they sometimes do irreparable damge to forest ecosystems but also the gases released due to these fires travel to the city causing health crises. While governments all over the world introduce initiatives to promote sustatinable practices and the general public abides by them, it is also important for scienitsts to know which forests are prone to fires. Through this project we intend to answer the question **_insert question_**.<br>


To help answer this question, we make use of the Algerian Forest Fire Data Set which has 244 observations of 11 variables and 1 output colummn, which in this case tells us whether there was a fire or not. Let's take a look at what these variables are and how they help us indicate the chances of a forest fire.**(this part will be written once Method is completely written to see how many words can still fit)**.

**Methods**<br>

**Expected outcomes and significance:**<br>
Through this project, we expect to find whether the following predictors **insert predictors** can help us determine **insert question** (elaborate more on that)<br>
As we, humans, fail to take actions against what causes these forest fires to happen so frequently, we need indicators which can help assist authorities in tackling this man-made disaster. Not only will these help firefighters understand whether there is a possibility of a forest fire passing onto a different region (given region's predictors) but also help scientists determine regions prone to fires.<br>
One important question after proecessing the entire data above would be whether there are more factors we can consider which can help understand nature of fires better or do we go on building better models with the given predictors.


(EDIT THAT)For this project, we make use of the Algerian Forest Fire Data Set which has 244 observations of 13 attributes and 1 output attribute, which in this case tells us whether there was a fire or not. <br>
<H4> Data set used: </H4>
Algerian forest fires: https://archive.ics.uci.edu/ml/machine-learning-databases/00547/Algerian_forest_fires_dataset_UPDATE.csv <br>
This data set has 14 variables, thirteen of which are numerical and one is categorical.

<H2> Loading the data set into R and wrangling it </H2>

In [1]:
install.packages("skimr")
install.packages("GGally")
library(tidyverse)
library(repr)
library(GGally)
library(tidymodels)
library(skimr)

Updating HTML index of packages in '.Library'

Making 'packages.html' ...
 done

Warning message in install.packages("GGally"):
“installation of package ‘GGally’ had non-zero exit status”
Updating HTML index of packages in '.Library'

Making 'packages.html' ...
 done

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──

✔ ggplot2 3.3.2     ✔ purrr   0.3.4
✔ tibble  3.0.3     ✔ dplyr   1.0.2
✔ tidyr   1.1.2     ✔ stringr 1.4.0
✔ readr   1.3.1     ✔ forcats 0.5.0

Warning message:
“package ‘ggplot2’ was built under R version 4.0.1”
Warning message:
“package ‘tibble’ was built under R version 4.0.2”
Warning message:
“package ‘tidyr’ was built under R version 4.0.2”
Warning message:
“package ‘dplyr’ was built under R version 4.0.2”
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()

Registered S3 method overwritten by 'GGally':
  method from   
  +.gg   gg

In [10]:
# URL that contains the dataframe that we want to examine
url <- "https://archive.ics.uci.edu/ml/machine-learning-databases/00547/Algerian_forest_fires_dataset_UPDATE.csv"

# Reading dataframe into R
algerian_forest_fires <- read_csv(url,skip = 1)

# Examining the data frame
# algerian_forest_data

# Inspecting the parsing failure
# algerian_forest_fires[168, 1]
algerian_forest_fires[170, 10] = '14.6'
algerian_forest_fires[170, 11] =  '9'
algerian_forest_fires[170, 12] =  '12.5'
algerian_forest_fires[170, 13] =  '10.4'
algerian_forest_fires[170, 14] =  "fire"

Parsed with column specification:
cols(
  day = col_character(),
  month = col_character(),
  year = col_character(),
  Temperature = col_character(),
  RH = col_character(),
  Ws = col_character(),
  Rain = col_character(),
  FFMC = col_character(),
  DMC = col_character(),
  DC = col_character(),
  ISI = col_character(),
  BUI = col_character(),
  FWI = col_character(),
  Classes = col_character()
)

Warning message:
“2 parsing failures.
row col   expected     actual                                                                                                       file
123  -- 14 columns 1 columns  'https://archive.ics.uci.edu/ml/machine-learning-databases/00547/Algerian_forest_fires_dataset_UPDATE.csv'
168  -- 14 columns 13 columns 'https://archive.ics.uci.edu/ml/machine-learning-databases/00547/Algerian_forest_fires_dataset_UPDATE.csv'
”


day month year Temperature RH Ws Rain FFMC DMC  DC     ISI  BUI  FWI  Classes
1 14  07    2012 37          37 18 0.2  88.9 12.9 14.6 9 12.5 10.4 fire NA

<H3> After a quick examination we can see that:</H3>
    <li> File that we read contains 2 dataframes.</li>
    <li> The collumn types of the data set are not parsed correctly.</li>
    <li> We know that there is something causing parsing issues in the rows : 123 and 168.</li> 
<p> To extract the target data frame, we would need to split the file into 2 dataframes that it contains, tidy each dataframe and join them </p>

<H4>Extracting and tidying Bejaja region data set</H4>

In [5]:
# Retrieving the bejaja data set
bejaja_forest <- slice(algerian_forest_fires, 2:122)

#Fixing the collumn types of the numerical variables
bejaja_forest_data <- bejaja_forest %>% 
    select(day:FWI) %>% 
    map_df(as.double) 

# Selecting the observation class data
bejaja_forest_fires <- bejaja_forest %>% 
    select(Classes) 

# Joining numerical variable and observation class data 
bejaja_forest_tidy <- cbind(bejaja_forest_data,bejaja_forest_fires)
#head(bejaja_forest_tidy)

<H4>Extracting and tidying Sidi-Bel region data set</H4>

In [7]:
#Finding the number of rows in the original dataset
number_of_rows <- nrow(algerian_forest_fires)

# Retrieving the bejaja data set
sidi_bel_forest <- slice(algerian_forest_fires, 125:number_of_rows)

gsub_convert <- function(x) {
    as.double(gsub(",",".",x))
}
#Fixing the collumn types of the numerical variables
sidi_bel_forest_data <- sidi_bel_forest %>% 
    select(day:FWI) %>% 
    map_df(gsub_convert)


# Selecting the observation class data
sidi_bel_forest_fires <- sidi_bel_forest %>% 
    select(Classes)

# Joining numerical variable and observation class data 
sidi_bel_forest_tidy <- cbind(sidi_bel_forest_data,sidi_bel_forest_fires)

Warning message in .f(.x[[i]], ...):
“NAs introduced by coercion”
Warning message in .f(.x[[i]], ...):
“NAs introduced by coercion”


<H4>Joining our two new dataframes</H4> 

In [8]:
algerian_forest_fires_tidy <- full_join(bejaja_forest_tidy, sidi_bel_forest_tidy)
algerian_forest_fires_tidy

# The resulting data frame is tidy and ready for the further analysis

Joining, by = c("day", "month", "year", "Temperature", "RH", "Ws", "Rain", "FFMC", "DMC", "DC", "ISI", "BUI", "FWI", "Classes")



day month year Temperature RH Ws Rain FFMC DMC  DC    ISI  BUI  FWI 
1    2  6     2012 29          61 13  1.3 64.4  4.1  7.6  1.0   3.9  0.4
2    3  6     2012 26          82 22 13.1 47.1  2.5  7.1  0.3   2.7  0.1
3    4  6     2012 25          89 13  2.5 28.6  1.3  6.9  0.0   1.7  0.0
4    5  6     2012 27          77 16  0.0 64.8  3.0 14.2  1.2   3.9  0.5
5    6  6     2012 31          67 14  0.0 82.6  5.8 22.2  3.1   7.0  2.5
6    7  6     2012 33          54 13  0.0 88.2  9.9 30.5  6.4  10.9  7.2
7    8  6     2012 30          73 15  0.0 86.6 12.1 38.3  5.6  13.5  7.1
8    9  6     2012 25          88 13  0.2 52.9  7.9 38.8  0.4  10.5  0.3
9   10  6     2012 28          79 12  0.0 73.2  9.5 46.3  1.3  12.6  0.9
10  11  6     2012 31          65 14  0.0 84.5 12.5 54.3  4.0  15.8  5.6
11  12  6     2012 26          81 19  0.0 84.0 13.8 61.4  4.8  17.7  7.1
12  13  6     2012 27          84 21  1.2 50.0  6.7 17.0  0.5   6.7  0.2
13  14  6     2012 30          78 20  0.5 59.0  4.6  7.8  1.0   4.4  0.4
14  15  6     2012 28          80 17  3.1 49.4  3.0  7.4  0.4   3.0  0.1
15  16  6     2012 29          89 13  0.7 36.1  1.7  7.6  0.0   2.2  0.0
16  17  6     2012 30          89 16  0.6 37.3  1.1  7.8  0.0   1.6  0.0
17  18  6     2012 31          78 14  0.3 56.9  1.9  8.0  0.7   2.4  0.2
18  19  6     2012 31          55 16  0.1 79.9  4.5 16.0  2.5   5.3  1.4
19  20  6     2012 30          80 16  0.4 59.8  3.4 27.1  0.9   5.1  0.4
20  21  6     2012 30          78 14  0.0 81.0  6.3 31.6  2.6   8.4  2.2
21  22  6     2012 31          67 17  0.1 79.1  7.0 39.5  2.4   9.7  2.3
22  23  6     2012 32          62 18  0.1 81.4  8.2 47.7  3.3  11.5  3.8
23  24  6     2012 32          66 17  0.0 85.9 11.2 55.8  5.6  14.9  7.5
24  25  6     2012 31          64 15  0.0 86.7 14.2 63.8  5.7  18.3  8.4
25  26  6     2012 31          64 18  0.0 86.8 17.8 71.8  6.7  21.6 10.6
26  27  6     2012 34          53 18  0.0 89.0 21.6 80.3  9.2  25.8 15.0
27  28  6     2012 32          55 14  0.0 89.1 25.5 88.5  7.6  29.7 13.9
28  29  6     2012 32          47 13  0.3 79.9 18.4 84.4  2.2  23.8  3.9
29  30  6     2012 33          50 14  0.0 88.7 22.9 92.8  7.2  28.3 12.9
30   1  7     2012 29          68 19  1.0 59.9  2.5  8.6  1.1   2.9  0.4
⋮   ⋮   ⋮     ⋮    ⋮           ⋮  ⋮  ⋮    ⋮    ⋮    ⋮     ⋮    ⋮    ⋮   
214  1  9     2012 29          86 16 0.0  37.9  0.9   8.2  0.1  1.4  0.0
215  2  9     2012 28          67 19 0.0  75.4  2.9  16.3  2.0  4.0  0.8
216  3  9     2012 28          75 16 0.0  82.2  4.4  24.3  3.3  6.0  2.5
217  4  9     2012 30          66 15 0.2  73.5  4.1  26.6  1.5  6.0  0.7
218  5  9     2012 30          58 12 4.1  66.1  4.0   8.4  1.0  3.9  0.4
219  6  9     2012 34          71 14 6.5  64.5  3.3   9.1  1.0  3.5  0.4
220  7  9     2012 31          62 15 0.0  83.3  5.8  17.7  3.8  6.4  3.2
221  8  9     2012 30          88 14 0.0  82.5  6.6  26.1  3.0  8.1  2.7
222  9  9     2012 30          80 15 0.0  83.1  7.9  34.5  3.5 10.0  3.7
223 10  9     2012 29          74 15 1.1  59.5  4.7   8.2  0.8  4.6  0.3
224 11  9     2012 30          73 14 0.0  79.2  6.5  16.6  2.1  6.6  1.2
225 12  9     2012 31          72 14 0.0  84.2  8.3  25.2  3.8  9.1  3.9
226 13  9     2012 29          49 19 0.0  88.6 11.5  33.4  9.1 12.4 10.3
227 14  9     2012 28          81 15 0.0  84.6 12.6  41.5  4.3 14.3  5.7
228 15  9     2012 32          51 13 0.0  88.7 16.0  50.2  6.9 17.8  9.8
229 16  9     2012 33          26 13 0.0  93.9 21.2  59.2 14.2 22.4 19.3
230 17  9     2012 34          44 12 0.0  92.5 25.2  63.3 11.2 26.2 17.5
231 18  9     2012 36          33 13 0.1  90.6 25.8  77.8  9.0 28.2 15.4
232 19  9     2012 29          41  8 0.1  83.9 24.9  86.0  2.7 28.9  5.6
233 20  9     2012 34          58 13 0.2  79.5 18.7  88.0  2.1 24.4  3.8
234 21  9     2012 35          34 17 0.0  92.2 23.6  97.3 13.8 29.4 21.6
235 22  9     2012 33          64 13 0.0  88.9 26.1 106.3  7.1 32.4 13.7
236 23  9     2012 35          56 14 0.0  89.0 29.4 115.6  7.5

In [9]:
# Setting a seed 
set.seed(2000)
# Splitting the data set
algerian_forest_fires_tidy <- mutate(algerian_forest_fires_tidy, Classes = as.factor(Classes))
forest_split <- initial_split(algerian_forest_fires_tidy, prop = 0.75, strata = Classes)
forest_train <- training(forest_split)
forest_test <- testing(forest_split)

# Specifying the summary function
my_skim <- skim_with(numeric = sfl(median = ~ median(., na.rm = TRUE), mean = ~ mean(., na.rm = TRUE),
                                   min = ~ min(., na.rm = TRUE), max = ~ max(., na.rm = TRUE)),
                                   append = FALSE)

# Creating summary table
summary_df <- my_skim(forest_train) %>% 
    tibble::as_tibble() %>% 
    select(skim_variable:numeric.max)
summary_df

skim_variable n_missing complete_rate factor.ordered factor.n_unique
1  Classes       1         0.9945355     FALSE           2             
2  day           0         1.0000000        NA          NA             
3  month         0         1.0000000        NA          NA             
4  year          0         1.0000000        NA          NA             
5  Temperature   0         1.0000000        NA          NA             
6  RH            0         1.0000000        NA          NA             
7  Ws            0         1.0000000        NA          NA             
8  Rain          0         1.0000000        NA          NA             
9  FFMC          0         1.0000000        NA          NA             
10 DMC           0         1.0000000        NA          NA             
11 DC            1         0.9945355        NA          NA             
12 ISI           0         1.0000000        NA          NA             
13 BUI           0         1.0000000        NA          NA             
14 FWI           1         0.9945355        NA          NA             
   factor.top_counts numeric.median numeric.mean numeric.min numeric.max
1  fir: 103, not: 79      NA                  NA     NA          NA     
2  NA                  16.00          16.0273224    1.0        31.0     
3  NA                   8.00           7.5519126    6.0         9.0     
4  NA                2012.00        2012.0000000 2012.0      2012.0     
5  NA                  32.00          32.1147541   24.0        40.0     
6  NA                  63.00          62.7868852   26.0        90.0     
7  NA                  15.00          15.5245902    6.0        29.0     
8  NA                   0.00           0.6595628    0.0        16.8     
9  NA                  83.70          77.6688525   28.6        94.3     
10 NA                  10.80          14.6289617    0.9        65.9     
11 NA                  31.25          49.4065934    6.9       220.4     
12 NA                   3.50           4.6672131    0.0        16.6     
13 NA                  11.50          16.6054645    1.4        68.0     
14 NA                   4.20           6.9758242    0.0        30.3

In [ ]:
# Removing the date data
options(repr.plot.width = 14, repr.plot.height= 14)
forest_train <- select(forest_train,Temperature:Classes)
num_data <- select(forest_train,Temperature:FWI)
ggpairs(forest_train, columns = c("Temperature","RH","Ws","Rain","FFMC","DMC","DC","ISI","BUI","FWI"),
        aes(colour = Classes)) 